integration-tests/src/test/java/org/zowe/apiml/functional/gateway/ServiceHaModeTest.java

```diff
@@ -11,89 +11,211 @@
+
+                        try {
+                            Response response = doRequest(method, url);
+                            if (response.getStatusCode() != HttpStatus.SC_OK) {
+                                fail();
+                            }
+                            StringTokenizer retryList = new StringTokenizer(response.getHeader("RibbonRetryDebug"), "|");
+                            assertThat(retryList.countTokens(), is(greaterThan(1)));
+                            break;
+                        } catch (RuntimeException | AssertionError e) {
+                            if (System.currentTimeMillis() - time0 > timeoutSec * 1000) throw e;
+                            await().timeout(1, TimeUnit.SECONDS);
+                        }
                     }
-                    StringTokenizer retryList = new StringTokenizer(response.getHeader("RibbonRetryDebug"), "|");
-                    assertThat(retryList.countTokens(), is(greaterThan(1)));
-                    break;
                 }
-                catch (RuntimeException | AssertionError e) {
-                    if (System.currentTimeMillis() - time0 > timeoutSec * 1000) throw e;
-                    await().timeout(1, TimeUnit.SECONDS);
+            }
+
+        }
+
+        @Nested
+        @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+        class OneReturns503 {
+
+            @BeforeAll
+            void setUp() throws LifecycleException, IOException {
+                List<Integer> ports = RandomPorts.generateUniquePorts(2);
+
+                service1 = new VirtualService("testHaModeService2", ports.get(0));
+                service2 = new VirtualService("testHaModeService2", ports.get(1));
+
+                service1.addInstanceServlet("Http503", "/httpCode");
+                service2.addHttpStatusCodeServlet(HttpStatus.SC_SERVICE_UNAVAILABLE);
+
+                service1.start();
+                service2.start().waitForGatewayRegistration(2, TIMEOUT);
+            }
+
+            @AfterAll
+            void cleanUp() {
+                try {
+                    service1.close();
+                } catch (Exception e) {
+                    // Just continue
+                }
+                try {
+                    service2.close();
+                } catch (Exception e) {
+                    // Just continue
                 }
             }
+
+            @ParameterizedTest
+            @MethodSource("org.zowe.apiml.functional.gateway.ServiceHaModeTest#retryableHttpMethods")
+            void verifyThatGatewayRetriesGet(Method method) {
+                routeAndVerifyRetries(service1.getGatewayUrls(), method, 2);
+            }
+
+            @ParameterizedTest
+            @MethodSource("org.zowe.apiml.functional.gateway.ServiceHaModeTest#nonRetryableHttpMethods")
+            void verifyThatGatewayNotRetriesPost(Method method) {
+                routeAndVerifyRetries(service1.getGatewayUrls(), method, 1);
+            }
```

Comment:
Is it possible to ad a test scenario for the connection refused exceptions? I suppose just calling a statically onboarded service with two instances, with one pointing to a port that is not set up should be enough to replicate

Reply:
Isn't this tested in `verifyThatGatewayRetriesToTheLiveOne` ?

Reply:
Yes, thank you, I missed that one. I see that the entire test class was disabled before.